src/lib/data/projects.ts

```diff
@@ -0,0 +1,93 @@
+import { PlanId } from '../../db/models/plan';
+import { ProjectId } from '../../db/models/project';
+import { Database } from '../../db/type';
+import { InstanceOfModel } from '../../db/util/types';
+import { isDefined, organizeObjectsByUniqueProperty } from '../../util';
+
+export interface ProjectData {
+  project: InstanceOfModel<Database['project']>;
+  projectVersion: InstanceOfModel<Database['projectVersion']>;
+  workflowStatus: InstanceOfModel<Database['workflowStatusOption']> | null;
+}
+
+export async function getAllProjectsForPlan({
+  database,
+  planId,
+}: {
+  database: Database;
+  planId: PlanId;
+  version: 'latest';
+}): Promise<Map<ProjectId, ProjectData>> {
+  const pvps = await database.projectVersionPlan.find({
+    where: {
+      planId,
+    },
+  });
+  const pvpsByProjectVersionId = organizeObjectsByUniqueProperty(
+    pvps,
+    'projectVersionId'
+  );
+  const projectVersions = await database.projectVersion.find({
+    where: (builder) =>
+      builder.whereIn('id', [
+        ...new Set(pvps.map((pvp) => pvp.projectVersionId)),
```

Comment:
If `projectVersionId` is a unique property of `pvps`, why put this map through `Set`? :)

You could use keys of `pvpsByProjectVersionId`
```suggestion
        ...pvpsByProjectVersionId.keys(),
```

Reply:
As far as I can tell, `projectVersionId` is actually not unique in `projectVersionPlan`, though it does form part of a primary key in conjunction with `planId`. I confirmed this by adding two rows with the same `projectVersionId` to a table but differing `planIds`.

That being said, logically a projectVersion should only ever be associated with a single plan, so this schema doesn't make sense as it currently stands, and `projectVersionId` isn't currently duplicated anywhere, so we could add a `UNIQUE` constraint to that column too.